lib/active_record_extended/patch/relation_patch.rb

```diff
@@ -4,16 +4,25 @@ module ActiveRecordExtended
   module Patch
     module RelationPatch
       module QueryDelegation
-        AR_EX_QUERY_METHODS = (
-          [
-            :with, :define_window, :select_window, :foster_select,
-            :either_join, :either_joins, :either_order, :either_orders
-          ] +
-          ActiveRecordExtended::QueryMethods::Unionize::UNIONIZE_METHODS +
-          ActiveRecordExtended::QueryMethods::Json::JSON_QUERY_METHODS
-        ).freeze
-
-        delegate(*AR_EX_QUERY_METHODS, to: :all)
+        BASE_QUERY_METHODS = [
+          :define_window, :select_window, :foster_select,
+          :either_join, :either_joins, :either_order, :either_orders
+        ].freeze
+
+        def self.query_methods
+          methods = BASE_QUERY_METHODS +
+                    ActiveRecordExtended::QueryMethods::Unionize::UNIONIZE_METHODS +
+                    ActiveRecordExtended::QueryMethods::Json::JSON_QUERY_METHODS
+
+          # Only include CTE methods if not disabled for Rails 7.2+
+          unless ActiveRecordExtended::CteDeprecationConfiguration.should_disable_cte?
+            methods << :with
+          end
+
+          methods
+        end
+
+        delegate(*query_methods, to: :all)
```

Comment:
This will not work for the same reason this doesn't work for the `prepend` this is executed before any config can be modified.